fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/services/ReplacePropertiesServiceImplTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl.services;
+
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.apache.jena.graph.NodeFactory.createURI;
+import static org.apache.jena.graph.NodeFactory.createLiteral;
+import static java.util.stream.Stream.of;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import org.apache.jena.datatypes.xsd.XSDDatatype;
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.Triple;
+import org.fcrepo.kernel.api.Transaction;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.operations.RdfSourceOperationFactory;
+import org.fcrepo.kernel.api.operations.RdfSourceOperationBuilder;
+import org.fcrepo.kernel.api.rdf.DefaultRdfStream;
+import org.fcrepo.kernel.impl.operations.UpdateRdfSourceOperation;
+import org.fcrepo.persistence.api.PersistentStorageSession;
+import org.fcrepo.persistence.api.PersistentStorageSessionManager;
+import org.fcrepo.persistence.api.exceptions.PersistentStorageException;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * DeleteResourceServiceTest
+ *
+ * @author bseeger
+ */
+@RunWith(MockitoJUnitRunner.Strict.class)
+public class ReplacePropertiesServiceImplTest {
+
+    @Mock
+    private Transaction tx;
+
+    @Mock
+    private PersistentStorageSession pSession;
+
+    @Mock
+    private PersistentStorageSessionManager psManager;
+
+    @Mock
+    private RdfSourceOperationFactory factory;
```

Comment:
this unit test could probably use real instances of the factories/operations, maybe draw the line at the persistence classes?

Reply:
The implementation classes for these interfaces are just stubs returning `null`, so unless we want to expand the scope we probably need this work done in a separate ticket and then pulled into the tests.
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/operations/RdfSourceOperationFactoryImpl.java

Reply:
I think that is in scope for this ticket, the ticket is for the workflow of replacing properties rather than for the service specifically

Reply:
Looks like RdfSourceOperationFactoryImpl was implemented, but it was not pulled into this test.